smolyay/surrogate.py

```diff
@@ -192,13 +192,23 @@ def gradient(self, x):
         """
         if self._coefficients is None:
             raise ValueError('Function needs training!')
-        x = numpy.array(x, copy=False, ndmin=1)
-        if x.shape != (self.dimension, ):
-            raise IndexError('Input must match dimension of domain.')
-        if not (numpy.all(x >= self._domain[:, 0])
-                and numpy.all(x <= self._domain[:, 1])):
+        x = numpy.array(x, copy=False, ndmin=2)
+        if self.dimension == 1:
+            if x.ndim == 2 and x.shape[0] == 1 and x.shape[1]  > 1:
+                # the cast to 2d puts these in wrong order, so transpose
+                x = x.T
+            else:
+                x = x[..., numpy.newaxis]
+        if x.shape[-1] != self.dimension:
+            raise IndexError("Input must match dimension of domain")
+        if self.dimension > 1:
+            oob = any(numpy.any(xi < bound[0]) or numpy.any(xi > bound[1]) 
+                    for xi,bound in zip(x.T,self.domain))
+        else:
+            oob = (numpy.any(x < self.domain[0]) or 
+                    numpy.any(x > self.domain[1]))
+        if oob:
             raise ValueError('x must lie in domain of surrogate')
-
         gradient = []
```

Comment:
Shouldn't `gradient` be a NumPy array rather than a list? You could preallocate it here, then assign the values below for each component.